File: flight_choice_prediction/v2/post/ios/FlightChoicePrediction.swift
import Amadeus

var amadeus: Amadeus = Amadeus(
    client_id: "YOUR_API_ID",
    client_secret: "YOUR_API_SECRET"
)

amadeus.shopping.flightOffersSearch.get(params: ["originLocationCode": "MAD",
                                                 "destinationLocationCode": "BER",
                                                 "departureDate": "2020-05-16",
                                                 "returnDate": "2020-05-30",
                                                 "adults": "2"],
                                        onCompletion: {
                                            response, error in
                                            if error == nil {
                                                amadeus.shopping.flightOffers.prediction.post(body: response!.result,
                                                                                              onCompletion: {
                                                                                                  prediction, _ in
                                                                                                  print(prediction!.data)
                                                })
                                            }
})

File: flight_choice_prediction/v2/post/Node SDK/flight_choice_prediction.js
const Amadeus = require("amadeus");

const amadeus = new Amadeus({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});

async function main() {
  try {
    const flightOffersResponse = await amadeus.shopping.flightOffersSearch.get({
      originLocationCode: "SYD",
      destinationLocationCode: "BKK",
      departureDate: "2022-08-01",
      adults: "2",
    });
    const response = await amadeus.shopping.flightOffers.prediction.post(
      flightOffersResponse
    );
    console.log(response.data);
  } catch (error) {
    console.error(error);
  }
}

main();
